README.md

```diff
@@ -40,6 +40,90 @@ esy test
 
 ### Running the node
 
+#### Deployment
+
+`esy x http_server` would run the node (currently over HTTP). `http_server` takes one argument - folder containing the config and data files.
+
+Example usage,
+
+```sh
+esy x http_server data/0
+```
+
+##### TLDR; If you can follow annotated shell scripts, checkout [setup.sh](./setup.sh), which contains the following steps as sequence of sh commands. You'll need [jq](https://github.com/stedolan/jq)
+
+To configure and setup the node, 
+1. Create identity for the node with `esy x sidecli generate-identity`. 
+   Step #1 will most likely be run multiple times for each node run by each participant. We create a list of identities of such authorised participants (validators), and use them in the next step - originating the consensus contract.
+2. `consensus.mligo` (found in `tezos_interop/`) needs to be deployed - a preloaded LIGO IDE with the contract can be found [here](https://ide.ligolang.org/p/-x6CdYJ5tIEaVzD9lGYsaA). Specify the list of validators' public addresses in `current_validators` field of the storage. Example,
+
+```ml
+{
+  root_hash = {
+    current_block_hash = 0x;
+    current_block_height = 0;
+    current_state_hash = 0x;
+    current_handles_hash = 0x;
+    current_validators = ([("edpk...": key)]: validators);
+  };
+  vault = {
+    known_handles_hash = (Big_map.empty : vault_known_handles_hash_set);
+    used_handles = (Big_map.empty : vault_used_handle_set);
+    vault = (Big_map.empty : vault);
+  }
+}
+```
+3. Use `setup-node` subcommand from `sidecli` to generate the config and the data files
+
+```sh
+    esy x sidecli setup-node "$data_directory" \
+	--secret=< \
+	--tezos_consensus_contract $contract_address_from_step_2 \
+	--tezos_rpc_node https://testnet-tezos.giganode.io \
+	--tezos_secret $test_net_wallet_secret \
+	--uri=$uri
+```
+
+**Obtaining the `tezos_secret` key
+This option takes the private key of the wallet to (TODO: do what?). While working with the test network, one needs to import the `secret` from the activate code JSON downloaded from [faucet.tzalpha.net](https://faucet.tzalpha.net) and activate the address.
```

Comment:
Should we add something about how to generate the key for `secret` too? That's the secret used by the validator that its address is derived from. 

  tezos-secret is one of the parameters in the json passed to [run_entrypoint.js](https://github.com/marigold-dev/sidechain/blob/0d41c92a9f9bec659c2eb7cbd9ecce703cf7e89c/tezos_interop/run_entrypoint.js#L14-L14) (specifically the parameter `secret`). The script uses it to sign a smart contract operation. 

In our use case, the script calls the entrypoint `update_root_hash` on the contract corresponding to the address passed in `tezos_consensus_contract`, and signs it with `tezos_secret`. 

@EduardoRFS what properties should the key passed for `tezos_secret` have? does it need to be configured with the mainchain smart contract in some way, or have money in its wallet to pay for gas, or something like that?

Reply:
> Should we add something about how to generate the key for secret too? 
The `tezos-client` commands that follow - they're suppose to answer this. Would you like me to clarify? What's not clear right now?

Reply:
Can I resolve this conversation?